src/components/Comment/component.js

```diff
@@ -75,34 +77,32 @@ export default class Comment extends React.Component {
 
     if (editing) return <CommentForm commentId={comment.id} close={closeEdit} />
 
-    const { image } = comment
-
     return <div className='comment' data-comment-id={comment.id}>
-      {canEditComment(currentUser, comment, community) &&
+      {notPending && canEditComment(currentUser, comment, community) &&
         <Dropdown alignRight toggleChildren={<Icon name='More' />}>
           {!image && <li><a onClick={edit}>Edit</a></li>}
           <li><a onClick={remove}>Remove</a></li>
         </Dropdown>}
-      <a name={`comment-${comment.id}`} />
+      {notPending && <a name={`comment-${comment.id}`} />}
       <Avatar person={person} showPopover />
       <div className='content'>
         {image && <div className='text'>
           <A to={`/u/${person.id}`} className='name'>{person.name}</A>
         </div>}
         {image && <a onClick={() => showImage(image.url, location.pathname, isMobile)}>
-          <img className='thumbnail' src={image.thumbnail_url} />
+          <ImageWithFallback className='thumbnail' preferredSrc={image.thumbnail_url} fallbackSrc={image.url} onlyUsePreferred={!fromTemp} />
```

Comment:
Maybe instead of adding another attribute, you can just omit the fallback when you know the thumbnail is going to be available, and then in the ImageWithFallback component, you can skip the fallback behavior if the fallback is not defined. I think that would make this component a bit easier to reason about and drop in wherever we want to have this "flicker-free" behavior in the future.